src/api.js

```diff
@@ -133,7 +133,11 @@ const Api = (options = {}) => {
     if (params.filter(p => p.position === 'path').some(p => request.indexOf(`{${p.nameOriginal}}`) === -1)) {
       throw new Error('Path Parameter not defined in given path.');
     }
+    if (params.filter(p => p.paramType === 'FieldsParam' && p.autoPrune === true).length > 1) {
```

Comment:
`FieldsParam` is the only param with this `paramType` attribute, is this for future consideration when doing the refactor? I understand why it's done, just wondering about the future